go-controller/pkg/ovn/kubevirt.go

```diff
@@ -110,18 +122,287 @@ func (oc *DefaultNetworkController) deleteDHCPOptions(pod *kapi.Pod) error {
 	return libovsdbops.DeleteDHCPOptionsWithPredicate(oc.nbClient, predicate)
 }
 
-func (oc *DefaultNetworkController) kubevirtCleanUp(pod *corev1.Pod) error {
-	if kubevirt.IsPodLiveMigratable(pod) {
-		isMigratedSourcePodStale, err := kubevirt.IsMigratedSourcePodStale(oc.watchFactory, pod)
+func (oc *DefaultNetworkController) deleteRoutingForMigratedPod(pod *kapi.Pod) error {
+	vm := kubevirt.ExtractVMNameFromPod(pod)
+	routePredicate := func(item *nbdb.LogicalRouterStaticRoute) bool {
+		return kubevirt.ExternalIDsContainsVM(item.ExternalIDs, vm)
+	}
+	if err := libovsdbops.DeleteLogicalRouterStaticRoutesWithPredicate(oc.nbClient, types.OVNClusterRouter, routePredicate); err != nil {
+		return fmt.Errorf("failed deleting pod routing when deleting the LR static routes: %v", err)
+	}
+	policyPredicate := func(item *nbdb.LogicalRouterPolicy) bool {
+		return kubevirt.ExternalIDsContainsVM(item.ExternalIDs, vm)
+	}
+	if err := libovsdbops.DeleteLogicalRouterPoliciesWithPredicate(oc.nbClient, types.OVNClusterRouter, policyPredicate); err != nil {
+		return fmt.Errorf("failed deleting pod routing when deleting the LR policies: %v", err)
+	}
+	return nil
+}
+
+func (oc *DefaultNetworkController) cleanUpForVM(pod *corev1.Pod) error {
+	isMigratedSourcePodStale, err := kubevirt.IsMigratedSourcePodStale(oc.watchFactory, pod)
+	if err != nil {
+		return fmt.Errorf("failed cleaning up VM when checking if pod is leftover: %v", err)
+	}
+	// Everything has already being cleand up since this is an old migration
+	// pod
+	if isMigratedSourcePodStale {
+		return nil
+	}
+	// This pod is not part of ip migration so we don't need to clean up
+	if !kubevirt.IsPodLiveMigratable(pod) {
+		return nil
+	}
+	if err := oc.deleteDHCPOptions(pod); err != nil {
+		return err
+	}
+	if err := oc.deleteRoutingForMigratedPod(pod); err != nil {
+		return err
+	}
+	return nil
+}
+
+// ensureLocalZonePodAddressesToNodeRoute will add static routes and policies to ovn_cluster_route logical router
+// to ensure VM traffic work as expected after live migration if the pod is running at the local/global zone
+// following is the list of NB logical resources created depending if it's interconnected or not:
+// IC:
+//   - static route with VM ip as dst-ip prefix and output port as node switch LRP
+//   - static route with cluster wide CIDR as src-ip prefix and nexthop GR, it has less
+//     priority than route to use overlay in case of pod to pod communication
+//
+// NO IC:
+//   - static route with VM ip as dst-ip prefix and output port as node switch LRP
+//   - low priority policy with src VM ip and reroute GR, since it has low priority
+//     it will not override the policy to enroute pod to pod traffic using overlay
```

Comment:
I don't think this description is accurate (or some other comments in the method). Please review it. Additionally, what is `node switch LRP`?

Reply:
I will fix comments, also note that we have decide to go back to cluster wide CIDR pointing to gw router since interconnect with multiple nodes is not going to be supported.